mlflow/utils/proto_json_utils.py

```diff
@@ -365,10 +366,15 @@ def parse_tf_serving_input(inp_dict, schema=None):
     import numpy as np
 
     def cast_schema_type(input_data):
+        input_data = deepcopy(input_data)
```

Comment:
why do we need to add deepcopy here?

Reply:
I think so. It changes input_data if it's a dictionary or list